### ISSUE TITLE
Make does not relink binaries when it should

### ISSUE THREAD

**Issue by Ganeti-Issues-Migrator** [ganeti/ganeti#968]
Originally reported of Google Code with ID 912.

```
<b>What steps will reproduce the problem?</b>
1. Change a string in src/GAneti/Query/Exec.hs
2. make src/hluxid

<b>What is the expected output? What do you see instead?</b>

Make should recompile Exec.o and then relink hluxid.
But make doesn't do the relink, so your changes are not reflected in the binary.
```
Originally added on 2014-08-07 12:24:11 +0000 UTC.

**Comment by Ganeti-Issues-Migrator** [comment]
```
-- Empty comment --
```
Originally added on 2014-08-07 12:25:46 +0000 UTC.

Added Labels: Type-Defect Priority-Medium
Changed State: Accepted
Added to Milestone: Release2.13

**Comment by Ganeti-Issues-Migrator** [comment]
```
Fixed in

commit 09ff6a256470d525218f707181f43a7e628fda58
Author: Yuto KAWAMURA(kawamuray) <kawamuray.dadada@gmail.com>
Date:   Thu Sep 25 04:10:44 2014 +0900

    Fix broken haskell dependencies
    
    Fixes issues 758 and 912.
    Since ghc -M does not generate inter object dependency rule for object
    files, dependencies from a target executable seed object
    (e.g. src/hluxid.o) to objects which finally will be linked to the
    target object (e.g. src/Ganeti/Daemon.o) are missing in Makefile.ghc.
    For instance, following commands will not rebuild src/hluxid inspite of
    its dependent object src/Ganeti/Daemon.o has been updated.
    
        $ make # rebuild all once
        $ touch src/Ganeti/Daemon.hs
        $ make # this will rebuild src/Ganeti/Daemon.o, not src/hluxid.o
    
    This patch adds simple substitution code after the Makefile.ghc
    generation to add additional inter object dependency for all object
    files to force relink related objects when a linked object has been
    updated.
    
    Signed-off-by: Yuto KAWAMURA(kawamuray) <kawamuray.dadada@gmail.com>
    Signed-off-by: Petr Pudlak <pudlak@google.com>
    Reviewed-by: Petr Pudlak <pudlak@google.com>

```
Originally added on 2014-10-20 15:24:16 +0000 UTC.

Changed State: Fixed
Added to Milestone: Release2.11
Removed from Milestone: Release2.13

**Comment by Ganeti-Issues-Migrator** [comment]
```
Released as part of 2.14.0
```
Originally added on 2015-06-02 10:39:05 +0000 UTC.

**Comment by Ganeti-Issues-Migrator** [comment]
```
Released as part of 2.14.0
```
Originally added on 2015-06-02 10:39:42 +0000 UTC.

Changed State: Released
